lib/backend/compat/compat.go

```diff
@@ -352,7 +353,7 @@ func ensureBlockAffinity(kvp *model.KVPair) *model.KVPair {
 		// Set AllocationBlock.HostAffinity to nil so it's never non-nil for the clients.
 		val.HostAffinity = nil
 	}
-	return &model.KVPair{Key: kvp.Key, Value: val}
+	return &model.KVPair{Key: kvp.Key, Value: val, Revision: kvp.Revision, TTL: kvp.TTL}
```

Comment:
Not sure TTL is valid to set on get / list operation?

https://github.com/projectcalico/libcalico-go/blob/master/lib/backend/model/keys.go#L90